pkg/autoops/api/api.go

```diff
@@ -727,6 +741,42 @@ func (s *AutoOpsService) UpdateAutoOpsRule(
 				}
 				return dt.Err()
 			}
+
+			// Delete a deletion schedule from the currently held schedules
+			extractDateTimeClauses, _ := autoOpsRule.ExtractDatetimeClauses()
+			for _, deleteClause := range req.DeleteClauseCommands {
+				delete(extractDateTimeClauses, deleteClause.Id)
+			}
+
+			// Check if there is a schedule with the same date and time.
+			for _, c := range req.AddDatetimeClauseCommands {
+				for _, extractDateTimeClause := range extractDateTimeClauses {
```

Comment:
Iterating `extractDateTimeClauses` in each validation is inefficient. How about creating a map with a time key at first and checking if the same time exists in each loop?

Reply:
I fixed https://github.com/bucketeer-io/bucketeer/pull/1171/commits/5f05b90543c7115fc2a7c6700de6e43e5508b044.